src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs

```diff
@@ -324,9 +319,14 @@ internal void LoadChildren(_ViewGroup panel)
 					HorizontalContentAlignment = HorizontalAlignment.Stretch,
 					IsTabStop = false
 				};
-
-				VisualTreeHelper.AddChild(this, FooterContentControl);
 			}
+
+			// We want FooterContentControl to be assigned before calling SetItemsPanel,
+			// because it may directly cause an ArrangeOverride to be called in some case,
+			// where it is expected to be non-null.
+			VisualTreeHelper.AddChild(this, HeaderContentControl);
```

Comment:
In addition to these changes in this PR, do we need to add more tests to prevent this -> 1st item fails to render for GridView

Reply:
i've commented in the issue that, i've failed to reproduce this in the context of runtime tests despite being a fairly simple setup...

Reply:
Thanks for the info, sorry did not see your comment on the issue before reviewing the PR